aws-py-static-website/__main__.py

```diff
@@ -0,0 +1,166 @@
+import json
+import mimetypes
+import os
+
+from pulumi import export, FileAsset, ResourceOptions, Config, Output
+import pulumi_aws
+import pulumi_aws.acm
+import pulumi_aws.cloudfront
+import pulumi_aws.config
+import pulumi_aws.route53
+import pulumi_aws.s3
+
+def get_domain_and_subdomain(domain):
+    parts = domain.split('.')
+    if len(parts) < 2:
+        raise Exception(f'No TLD found on ${domain}')
+    if len(parts) == 2:
+        return "", domain
+    subdomain = parts[0]
+    parts.pop(0)
+    return subdomain, '.'.join(parts) + '.'
+
+stack_config = Config('static-website')
+target_domain = stack_config.require('targetDomain')
+path_to_website_contents = stack_config.require('pathToWebsiteContents')
+certificate_arn = stack_config.get('certificateArn')
+
+content_bucket = pulumi_aws.s3.Bucket('contentBucket',
+    bucket=target_domain,
+    acl='public-read',
+    website={
+        'index_document': 'index.html',
+        'error_document': '404.html'
+    })
+
+def crawl_directory(content_dir, f):
+    for file in os.listdir(content_dir):
+        filepath = os.path.join(content_dir, file)
+
+        if os.path.isdir(filepath):
+            crawl_directory(filepath, f)
+        elif os.path.isfile(filepath):
+            f(filepath)
+
+web_contents_root_path = os.path.join(os.getcwd(), path_to_website_contents)
+def bucket_object_converter(filepath):
+    relative_path = filepath.replace(web_contents_root_path + '/', '')
+    mime_type, _ = mimetypes.guess_type(filepath)
+    content_file = pulumi_aws.s3.BucketObject(
+        relative_path,
+        key=relative_path,
+        acl='public-read',
+        bucket=content_bucket,
+        content_type=mime_type,
+        source=FileAsset(filepath),
+        opts=ResourceOptions(parent=content_bucket)
+    )
+
+crawl_directory(web_contents_root_path, bucket_object_converter)
+
+logs_bucket = pulumi_aws.s3.Bucket('requestLogs', bucket=f'{target_domain}-logs', acl='private')
+
+TEN_MINUTES = 60 * 10
+
+# provision a certificate if the arn is not provided
```

Comment:
The hill I have chosen to die on is that [Comments Should be Sentences](https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html). So here, and else where, should be complete sentences starting with a capital letter and ending with a period.

I believe I am the only one at the company who feels strongly about this, so feel free to take this feedback as any other matter of opinion.

Reply:
That's fine. I think for an example, you're right that we should have clearer comments.